File: src/server/RoofLights.server.lua
local ServerStorage = game:GetService("ServerStorage")
local CollectionService = game:GetService("CollectionService")
local Transparency = ServerStorage.RoofLights.Transparency
local Brightness = ServerStorage.RoofLights.Brightness
local Colour = ServerStorage.RoofLights.Colour
local RoofLightControls = game.Workspace.Studio.RoofLightControls
local RoofLights: { Part } = CollectionService:GetTagged("RoofLight")
local RoofLightManager = require(ServerStorage.TS.RoofLightManager)

RoofLightControls.ToggleLight.Button.ProximityPrompt.Triggered:Connect(function()
	RoofLightManager.Toggle()
end)

RoofLightControls.FadeLightToggle.Button.ProximityPrompt.Triggered:Connect(function()
	if Brightness.Value == 4 then
		RoofLightManager.FadeOut()
	else
		RoofLightManager.FadeIn()
	end
end)

Brightness.Changed:Connect(function(value)
	for _, light in ipairs(RoofLights) do
		local SurfaceLight = light:FindFirstChildWhichIsA("SurfaceLight")
		SurfaceLight.Brightness = value
	end
end)

Colour.Changed:Connect(function(value)
	for _, light in ipairs(RoofLights) do
		local SurfaceLight = light:FindFirstChildWhichIsA("SurfaceLight")
		SurfaceLight.Color = value
	end
end)

Transparency.Changed:Connect(function(value)
	for _, light in ipairs(RoofLights) do
		light.Transparency = value
	end
end)

File: src/server/RoofLights.server.ts
import { CollectionService, ServerStorage } from "@rbxts/services";
import RoofLightManager from "serverStorage/RoofLightManager";

const RoofLightControls = game.Workspace.Studio.RoofLightControls;
const { Transparency, Brightness, Colour } = ServerStorage.RoofLights;
const RoofLights: Part[] = [];

CollectionService.GetTagged("RoofLight").forEach((item) => {
	if (item.IsA("Part")) {
		RoofLights.push(item);
	}
});

RoofLightControls.ToggleLight.Button.ProximityPrompt.Triggered.Connect(() => {
	RoofLightManager.Toggle();
});

RoofLightControls.FadeLightToggle.Button.ProximityPrompt.Triggered.Connect(() => {
	if (Brightness.Value === 4) {
		RoofLightManager.FadeOut();
	} else {
		RoofLightManager.FadeIn();
	}
});

Brightness.Changed.Connect((value) => {
	RoofLights.forEach((light) => {
		const SurfaceLight = light.FindFirstChildWhichIsA("SurfaceLight") as SurfaceLight;
		SurfaceLight.Brightness = value;
	});
});

Colour.Changed.Connect((value) => {
	RoofLights.forEach((light) => {
		const SurfaceLight = light.FindFirstChildWhichIsA("SurfaceLight") as SurfaceLight;
		SurfaceLight.Color = value;
	});
});

Transparency.Changed.Connect((value) => {
	RoofLights.forEach((light) => {
		light.Transparency = value;
	});
});
